### PR TITLE
[SERVICES-226] Http status codes

### PR DESCRIPTION
**Pull request by ArturKlajnerok**
Http status codes for all resources and tests for them. Also changed Redis storage to not return internal errors when specific key doesn't exist.

Take a look: @pchojnacki @jcellary @michalroszka

### CODE REVIEW

File: helios/oauth_controller.go [hunk 1]
```diff
@@ -53,6 +53,12 @@ func (controller *OAuthController) infoHandler(w http.ResponseWriter, r *http.Re
 	if ir := controller.server.HandleInfoRequest(resp, r); ir != nil {
 		controller.server.FinishInfoRequest(resp, r, ir)
 		resp.Output["user_id"] = ir.AccessData.UserData
+	} else {
+		if resp.InternalError != nil {
+			resp.StatusCode = http.StatusInternalServerError
+		} else {
```
Comment (aquilax) [inline helios/oauth_controller.go]: You don't need else here :no_bell:

Comment (nandy-andy) [inline helios/oauth_controller.go]: Hey mate! :wave: Bored in Sweden, heh? :japanese_ogre:

Comment (aquilax) [inline helios/oauth_controller.go]: Ha ha a little. But my point is still valid :palm_tree:

Comment (ArturKlajnerok) [inline helios/oauth_controller.go]: Hey @aquilax ;), if handling info request fails and there's no internal error my default http status is "http.StatusBadRequest", so this else fits for that conditions :snail:

Comment (aquilax) [inline helios/oauth_controller.go]: @ArturKlajnerok the way I see it `resp.InternalError` is either nil or not so there are only two options for `resp.StatusCode` in this block `http.StatusInternalServerError` | `http.StatusBadRequest`

So this should be equivalent:
```go
resp.StatusCode = http.StatusBadRequest
if resp.InternalError != nil {
  resp.StatusCode = http.StatusInternalServerError
}
```

Cheers! :dizzy_face:

Comment (ArturKlajnerok) [inline helios/oauth_controller.go]: Yeah that make sense :panda_face:

**Comment by jcellary** [comment]
LGTM

**Comment by pchojnacki** [comment]
LGTM

**Pull request merged**